beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/path/aws/S3DryRunPathCleanerTest.java

```diff
@@ -45,23 +45,29 @@
   private final String partition1Sentinel = "table/id1/partition_1_$folder$";
   private final String absolutePath = "s3://" + bucket + "/" + keyRoot;
   private final String tableName = "table";
+  private final String databaseName = "database";
 
   private final S3Mock s3Mock = new S3Mock.Builder().withPort(0).withInMemoryBackend().build();
+  private EntityHousekeepingPath housekeepingPath;
   private AmazonS3 amazonS3;
   private S3Client s3Client;
-  private S3BytesDeletedReporter s3BytesDeletedReporter;
-  private @Mock MeterRegistry meterRegistry;
+  private @Mock S3BytesDeletedReporter s3BytesDeletedReporter;
 
   private S3PathCleaner s3DryRunPathCleaner;
 
   @BeforeEach
   void setUp() {
-    when(meterRegistry.counter(anyString())).thenReturn(mock(Counter.class));
     amazonS3 = AmazonS3Factory.newInstance(s3Mock);
     amazonS3.createBucket(bucket);
     s3Client = new S3Client(amazonS3, true);
-    s3BytesDeletedReporter = new S3BytesDeletedReporter(s3Client, meterRegistry, false);
     s3DryRunPathCleaner = new S3PathCleaner(s3Client, new S3SentinelFilesCleaner(s3Client), s3BytesDeletedReporter);
+    housekeepingPath = new EntityHousekeepingPath.Builder()
```

Comment:
This is not that important, but I quite like when I can use the Builder pattern combined with a static factory method.

```java
housekeepingPath = EntityHousekeepingPath.builder()
  .path(...)
  ...
  .build();
```

The `builder` method returns a new instance of `Builder`.

Reply:
>  Builder pattern combined with a static factory method.

Not sure what you mean by this, what would this look like? thanks!

Reply:
Discussed in person. Now the example makes sense.

Reply:
I see what you mean, will add this to my list of useful refactorings :) thanks